### DISCUSSION TITLE
Thoughts on asdf as a possible toolchain provider

### DISCUSSION THREAD

**Discussion by andrejbaran** [moonrepo/moon#1198] [Ideas]
Hey 👋 

I was wondering what are your thoughts on intergration of [asdf](https://asdf-vm.com/) as an alternative toolchain provider?

It's a version manager for various toolchains (Rust, Go, PHP, Node, Bun, Deno, ....) 

Thanks!

**Comment by milesj** [comment]
We built our own toolchain (proto) because there was nothing available that was truly cross-platform and performant. asdf is Unix only, and is written entirely with Bash scripts.

We needed something cross-platform, and ideally in Rust. Because of that, we have no plans to ever support another toolchain.

**Reply by andrejbaran** [reply]
Thanks for the quick response. That makes sense.

**Reply by airtonix** [reply]
you should probably rethink this.

somthing like running the asdf bash plugins in wasm would instantly give access to hundreds of plugins

**Reply by milesj** [reply]
There's a bounty for it: https://github.com/moonrepo/tools/issues/9
